### PR TITLE
Update Export Guide

### PR DESCRIPTION
**Pull request by jenissabarrera**
update export guide, add new instructions for regex filter, etc...

### CODE REVIEW

**Comment by dginty4** [comment]
@jenissabarrera You'll need to run go generate just to sync the docs

File: examples/resources/genesyscloud_tf_export/resource.tf [hunk 1]
```diff
@@ -1,9 +1,79 @@
...
+
+# This will only generate group resources with the string ending with "dev" or "test"
+resource "genesyscloud_tf_export" "include-filter" {
+  directory                = "./genesyscloud/include-filter"
+  export_as_hcl            = true
+  log_permission_errors    = true
   include_state_file       = true
-  exclude_attributes       = ["genesyscloud_user.skills"]
+  include_filter_resources = ["genesyscloud_group::-(agent|jen)$"]
+  # include_filter_resources = ["genesyscloud_routing_queue::Barrera Test Queue"]
```
Comment (dginty4) [inline examples/resources/genesyscloud_tf_export/resource.tf:29]: Is this line supposed to be here? Same with the one below.

File: examples/resources/genesyscloud_tf_export/resource.tf [hunk 2]
```diff
@@ -1,9 +1,79 @@
...
+
+}
+
+# This will generate ALL resources in the org except the ones listed
+resource "genesyscloud_tf_export" "exclude-filter" {
+  directory             = "./genesyscloud/exclude-filter"
+  export_as_hcl         = true
+  log_permission_errors = true
+  include_state_file    = true
+  exclude_filter_resources = [
```
Comment (dginty4) [inline examples/resources/genesyscloud_tf_export/resource.tf:40]: I don't think this list needs to be so long, people will understand the purpose of the attribute with a few items

File: examples/resources/genesyscloud_tf_export/resource.tf [hunk 3]
```diff
@@ -1,9 +1,79 @@
...
+
+
+terraform {
+  required_providers {
+    genesyscloud = {
+      source  = "mypurecloud/genesyscloud"
+      version = "1.32.0"
+    }
+  }
+}
```
Comment (charliecon) [inline examples/resources/genesyscloud_tf_export/resource.tf:19]: This terraform block can be omitted, along with the commented resource above

File: docs/guides/export.md [hunk 4]
```diff
@@ -29,4 +29,61 @@ Once your export resource is configured, run `terraform init` to set up Terrafor
...
+
+If you want to include resources that begin or end with “dev” or “test”, use the following format:
+
+
+```hcl
+resource "genesyscloud_tf_export" "include-filter" {
+  directory = "./genesyscloud/include-filter"
+  export_as_hcl = true
+  log_permission_errors = true
+  include_filter_resources = ["genesyscloud_group::.*(?:dev|test)$"]
```
Comment (charliecon) [inline docs/guides/export.md:48]: Could you line up all the equals characters vertically? The resource is totally valid the way you have it, but it's much tidier looking when they're lined up. Same with the other resource below

**Review by dginty4 — APPROVED** [review]

**Comment by jenissabarrera** [comment]
Hi @HemanthDogiparthi12 and @charliecon, changes have been made. Let me know if there's anything else that needs to be modified.

**Comment by charliecon** [comment]
Hi @jenissabarrera 

I just noticed you edited the guide inside the docs folder. I think you should be be making the changes inside `templates/guides/export.md.tmpl` and running `go generate` to update the files inside `docs`

**Review by carnellj-genesys — APPROVED** [review]

**Review by HemanthDogiparthi12 — APPROVED** [review]

**Comment by jenissabarrera** [comment]
Hi @charliecon, done with this.

**Comment by charliecon** [comment]
> Hi @charliecon, done with this.

Are you not going to include the content you added to the export guide? Here it is in a previous commit - [https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/971/commits/85a7a1199c70b44dc8a673058cde6ff89c6a0639](https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/971/commits/85a7a1199c70b44dc8a673058cde6ff89c6a0639)

You might have missed a comment I left before that said you need to add those changes to `templates/guides/export.md.tmpl` , and not `docs/guides/export.md`. Everything inside `/docs/` is generated based on schema definitions and the content inside `/templates/` and `/examples/`

**Comment by jenissabarrera** [comment]
Hi @charliecon, Yes, I missed that. Thank you for pointing that out. Done changing the export.md.tmpl file.

File: templates/guides/export.md.tmpl [hunk 5]
```diff
@@ -29,4 +29,61 @@ Once your export resource is configured, run `terraform init` to set up Terrafor
...
+
+On the other hand, Terraform also provides the `exclude_attributes` option for instances where certain fields need to be omitted from an export. This, along with the ability to automatically export additional dependencies, contributes to Terraform’s flexible framework for managing resource exports. It allows for granular control over the inclusion or exclusion of elements in the export, ensuring that your exported configuration aligns precisely with your requirements.
+
 If exported resources contain references to objects that we don't intend to manage with Terraform or if they cannot be resolved using an API call then a variable will be generated to refer to that object. A definition for that variable will be provided in a generated `terraform.tfvars` file. The reference variables must be filled out with the values of the corresponding resources in a different org before being applied to it.
```
Comment (charliecon) [inline templates/guides/export.md.tmpl:89]: This paragraph is duplicated here on line 89 and line 32. Once this is fixed, you'll need to run `go generate` to render these changes to the guide inside `/docs/`

**Review by charliecon — APPROVED** [review]

**Pull request merged**